src/App/Fossa/ReleaseGroup/Create.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.ReleaseGroup.Create (
+  createMain,
+  retrievePolicyId,
+  retrieveTeamIds,
+) where
+
+import App.Fossa.Config.ReleaseGroup.Create (CreateConfig (..))
+import App.Fossa.ReleaseGroup.Common (retrieveReleaseGroupId)
+import App.Types (ReleaseGroupRevision (..))
+import Control.Algebra (Has)
+import Control.Effect.Diagnostics (Diagnostics, fatalText)
+import Control.Effect.FossaApiClient (FossaApiClient, createReleaseGroup, getPolicies, getReleaseGroups, getTeams)
+import Control.Effect.Lift (Lift)
+import Control.Monad (when)
+import Data.Map qualified as Map
+import Data.Maybe (isJust, mapMaybe)
+import Data.String.Conversion (ToText (..))
+import Data.Text (Text, intercalate)
+import Effect.Logger (Logger, logInfo, logStdout)
+import Fossa.API.Types (CreateReleaseGroupRequest (..), CreateReleaseGroupResponse (..), Policy (..), PolicyType (..), Team (..))
+
+createMain ::
+  ( Has Diagnostics sig m
+  , Has Logger sig m
+  , Has (Lift IO) sig m
+  , Has FossaApiClient sig m
+  ) =>
+  CreateConfig ->
+  m ()
+createMain CreateConfig{..} = do
+  logInfo "Running FOSSA release-group create"
+
+  releaseGroups <- getReleaseGroups
+  maybeReleaseGroupId <- retrieveReleaseGroupId (releaseGroupTitle releaseGroupRevision) releaseGroups
+  -- Having release groups with the same name is a current functionality. However, we want to refrain from creating release groups with the same
+  -- name on the CLI as it will make it harder to determine which release groups users want to modify in our other release group commands.
+  when (isJust maybeReleaseGroupId) $
+    fatalText $
+      "Release Group `" <> releaseGroupTitle releaseGroupRevision <> "` already exists"
+
+  policies <- getPolicies
+  teams <- getTeams
+
+  maybeLicensePolicyId <- retrievePolicyId (releaseGroupLicensePolicy releaseGroupRevision) LICENSING policies
+  maybeSecurityPolicyId <- retrievePolicyId (releaseGroupSecurityPolicy releaseGroupRevision) SECURITY policies
```

Comment:
[nit] you should log these steps, right now in cmdline all we see is, for long time while 5 or 6 requests are performed sequentially. You can use `StickyLogger` or `context` 

```
Running FOSSA release-group create
```